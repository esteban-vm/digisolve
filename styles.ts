import { css } from '@emotion/react'

export const globalStyles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;

    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;

    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;

    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;

    --color-shadow: #efefefef;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
