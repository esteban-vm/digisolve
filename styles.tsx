import { Global, css } from '@emotion/react'

const styles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --primary-color: #ed07b7;
    --primary-color-dark: #bd0592;
    --secondary-color: #6920b0;
    --light-color: #acdaff;
    --dark-color: #555;
  }

  html,
  body {
    background-color: #fff;
    color: #000;
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }
`

export const Globals = () => <Global styles={styles} />
export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
