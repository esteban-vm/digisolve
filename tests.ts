import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
