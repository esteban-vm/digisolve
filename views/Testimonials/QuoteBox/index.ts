export { default, type QuoteProps } from './QuoteBox'
