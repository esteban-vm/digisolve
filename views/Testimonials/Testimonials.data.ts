import type { TestimonialProps } from './Testimonial'
import { faker } from '@faker-js/faker'
import { uuid } from '@/helpers'

const content = faker.lorem.sentence(8)
const width = 72
const height = 72

const testimonials: TestimonialProps[] = [
  {
    id: uuid(),
    customer: faker.name.fullName({ sex: 'male' }),
    content,
    image: { src: '/img/client-1.jpg', width, height },
  },
  {
    id: uuid(),
    customer: faker.name.fullName({ sex: 'male' }),
    content,
    image: { src: '/img/client-2.jpg', width, height },
  },
  {
    id: uuid(),
    customer: faker.name.fullName({ sex: 'female' }),
    content,
    image: { src: '/img/client-3.jpg', width, height },
  },
]

export default testimonials
