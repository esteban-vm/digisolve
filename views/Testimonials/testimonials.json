[
  {
    "image": "/img/client-1.jpg",
    "text": "Aaron Cruz",
    "height": 72,
    "width": 72
  },
  {
    "image": "/img/client-2.jpg",
    "text": "Jeff Jones",
    "height": 72,
    "width": 72
  },
  {
    "image": "/img/client-3.jpg",
    "text": "Helen Smith",
    "height": 72,
    "width": 72
  }
]
