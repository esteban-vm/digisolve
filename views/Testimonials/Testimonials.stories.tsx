import type { Meta } from '@storybook/react'
import Testimonials from './Testimonials'

export default {
  component: Testimonials,
  title: 'Views/Testimonials',
} as Meta<typeof Testimonials>

export const TestimonialsView = () => <Testimonials />
