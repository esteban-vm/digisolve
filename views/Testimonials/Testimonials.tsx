import type { FunctionalComponent } from '@/types'
import { client1, client2, client3 } from '@/utils/images'
import { styled } from '@/utils/styles'
import QuoteBox, { type QuoteProps } from './QuoteBox'

export const quotes: QuoteProps[] = [
  { name: 'Aaron Cruz', photo: client1 },
  { name: 'Jeff Jones', photo: client2 },
  { name: 'Helen Smith', photo: client3 },
]

const TestimonialsComponent: FunctionalComponent = (props) => {
  return (
    <section {...props}>
      <div className='row'>
        <h2>Our Testimonials</h2>
      </div>
      <div className='row'>
        {quotes.map((quote, index) => (
          <QuoteBox key={index} {...quote} role='listitem' />
        ))}
      </div>
    </section>
  )
}

const Testimonials = styled(TestimonialsComponent)`
  background-image: linear-gradient(var(--color-overlay3), var(--color-overlay4)), url('/img/testimonial-bg.jpg');
  background-size: cover;
  background-attachment: fixed;
  text-align: center;
  color: var(--color-white);

  h2 {
    font-size: 250%;
    margin-bottom: 20px;
    margin-top: 150px;
  }
`

export default Testimonials
