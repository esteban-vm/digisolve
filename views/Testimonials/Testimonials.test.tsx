import { render, cleanup, screen, create } from '@/tests'
import { mediaQuery } from '@/styles'
import testimonials from './Testimonials.data'
import Testimonials from './Testimonials'

describe('ðŸ§ª TESTIMONIALS:', () => {
  describe('display tests:', () => {
    /** Section title */
    const name = /^our testimonials$/i

    beforeEach(() => {
      render(<Testimonials />)
    })

    afterEach(cleanup)

    it('should be accessible', () => {
      const section = screen.getByRole('region', { name })
      expect(section).toBeInTheDocument()
    })

    describe('should be displayed:', () => {
      it('the title', () => {
        const title = screen.getByRole('heading', { name, level: 2 })
        expect(title).toBeInTheDocument()
        expect(title).toBeVisible()
      })

      it('the testimonials', () => {
        const testimonialElements = screen.getAllByRole('article')
        expect(testimonialElements).toHaveLength(testimonials.length)

        for (const testimonial of testimonialElements) {
          expect(testimonial).toBeInTheDocument()
          expect(testimonial).toBeVisible()
        }
      })
    })
  })

  describe('style tests:', () => {
    let tree: ReturnType<typeof create>

    beforeEach(() => {
      tree = create(<Testimonials />)
    })

    it('should render with correct styles', () => {
      expect(tree).toMatchSnapshot()
    })

    describe('should have correct styles on:', () => {
      it('extra small screen devices', () => {
        expect(tree).toHaveStyleRule('padding', '0 10%', { media: mediaQuery('xs') })
      })

      it('small screen devices', () => {
        const media = mediaQuery('sm')
        const target = 'h2'
        expect(tree).toHaveStyleRule('margin-top', '60px', { target, media })
        expect(tree).toHaveStyleRule('margin-bottom', '10px', { target, media })
      })

      it('large screen devices', () => {
        expect(tree).toHaveStyleRule('margin-top', '90px', { target: 'h2', media: mediaQuery('lg') })
      })
    })
  })
})
