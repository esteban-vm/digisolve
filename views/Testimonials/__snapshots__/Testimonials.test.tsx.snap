// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª TESTIMONIALS: style tests: should render with correct styles 1`] = `
.emotion-0 {
  background-image: linear-gradient(var(--color-overlay3), var(--color-overlay4)),url('/img/testimonial-bg.jpg');
  -webkit-background-size: cover;
  background-size: cover;
  background-attachment: fixed;
  text-align: center;
  color: var(--color-white);
}

@media only screen and (max-width: 480px) {
  .emotion-0 {
    padding: 0 10%;
  }
}

.emotion-0 h2 {
  font-size: 250%;
  margin-bottom: 20px;
  margin-top: 150px;
  text-transform: capitalize;
}

@media only screen and (max-width: 1200px) {
  .emotion-0 h2 {
    margin-top: 90px;
  }
}

@media only screen and (max-width: 768px) {
  .emotion-0 h2 {
    margin-top: 60px;
    margin-bottom: 10px;
  }
}

.emotion-1 {
  text-align: left;
  margin-bottom: 50px;
}

@media only screen and (max-width: 480px) {
  .emotion-1 {
    margin-bottom: 5px;
  }
}

.emotion-1 blockquote {
  padding: 2%;
  font-style: italic;
  line-height: 145%;
  position: relative;
  margin-top: 40px;
}

.emotion-1 blockquote::before {
  content: '\\201C';
  display: block;
  font-size: 500%;
  position: absolute;
  left: -5px;
  top: -5px;
}

.emotion-1 blockquote::first-letter {
  text-transform: capitalize;
}

.emotion-1 cite {
  display: block;
  margin-top: 25px;
  font-size: 90%;
  text-transform: capitalize;
}

.emotion-1 img {
  border-radius: 50%;
  height: 20%;
  vertical-align: middle;
  margin-right: 10px;
}

<section
  aria-labelledby=":r0:"
  className="emotion-0"
  id="testimonials"
>
  <div
    className="row "
    role="presentation"
  >
    <h2
      id=":r0:"
    >
      Our testimonials
    </h2>
  </div>
  <div
    className="row "
    role="presentation"
  >
    <div
      className="col span_1_of_3 "
      role="presentation"
    >
      <article
        aria-labelledby="9a4de570-5383-4d55-9e65-834cf387e987"
        className="emotion-1"
      >
        <blockquote>
          Nam consequuntur atque illum aliquam dolor illo eveniet.
        </blockquote>
        <cite>
          <img
            alt="Merle Hyatt"
            data-nimg="1"
            decoding="async"
            height={72}
            id="9a4de570-5383-4d55-9e65-834cf387e987"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2Fclient-1.jpg&w=256&q=75"
            srcSet="/_next/image?url=%2Fimg%2Fclient-1.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fclient-1.jpg&w=256&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={72}
          />
          Merle Hyatt
        </cite>
      </article>
    </div>
    <div
      className="col span_1_of_3 "
      role="presentation"
    >
      <article
        aria-labelledby="fb12dec5-b624-4166-8fcf-373add8499ef"
        className="emotion-1"
      >
        <blockquote>
          Nam consequuntur atque illum aliquam dolor illo eveniet.
        </blockquote>
        <cite>
          <img
            alt="Marcus Lakin"
            data-nimg="1"
            decoding="async"
            height={72}
            id="fb12dec5-b624-4166-8fcf-373add8499ef"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2Fclient-2.jpg&w=256&q=75"
            srcSet="/_next/image?url=%2Fimg%2Fclient-2.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fclient-2.jpg&w=256&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={72}
          />
          Marcus Lakin
        </cite>
      </article>
    </div>
    <div
      className="col span_1_of_3 "
      role="presentation"
    >
      <article
        aria-labelledby="c5c7ea89-2276-41d0-9a08-bebcb9fa4edc"
        className="emotion-1"
      >
        <blockquote>
          Nam consequuntur atque illum aliquam dolor illo eveniet.
        </blockquote>
        <cite>
          <img
            alt="Misty Cronin"
            data-nimg="1"
            decoding="async"
            height={72}
            id="c5c7ea89-2276-41d0-9a08-bebcb9fa4edc"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2Fclient-3.jpg&w=256&q=75"
            srcSet="/_next/image?url=%2Fimg%2Fclient-3.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg%2Fclient-3.jpg&w=256&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={72}
          />
          Misty Cronin
        </cite>
      </article>
    </div>
  </div>
</section>
`;
